### ISSUE TITLE
Error on using approximation for Betweenness

### ISSUE THREAD

**Issue by apitts** [networkx/networkx#1860]
When entering an integer for k in the calculation of betweenness I receive the following error message:

```
File "/home/andrew/miniconda3/envs/polinode-metrics/lib/python3.4/site-packages/networkx/algorithms/centrality/betweenness.py", line 112, in betweenness_centrality
    nodes = random.sample(G.nodes(), k)
  File "/home/andrew/miniconda3/envs/polinode-metrics/lib/python3.4/random.py", line 311, in sample
    raise TypeError("Population must be a sequence or set.  For dicts, use list(d).")
TypeError: Population must be a sequence or set.  For dicts, use list(d).
```

Any thoughts? It would seem that random.sample requires G.nodes( ) be wrapped in a list( ).

**Comment by apitts** [comment]
To reproduce please try:
```Python
G=nx.complete_graph(5)
b=nx.betweenness_centrality(G, k=2, weight=None, normalized=False, endpoints=True)
```

**Comment by dschult** [comment]
That example works with the latest dev version.  It also works with the v1.10 stable version.  But between those two the behavior of G.nodes() changed from returning a list to returning an iterator and random_sample was broken for a short time.  I suspect you got a version from github in summer that hasn't been updated to the latest dev. Can you check that? If so, update to latest master and try again.

**Comment by apitts** [comment]
Thanks @dschult! Yes, I was using the master branch. The issue is resolved when v1.10 is used.
